### PR TITLE
Add support for git tags when publishing.

### PR DESCRIPTION
**Pull request by ahmedcharles**
#220

### CODE REVIEW

File: publish.sh [hunk 1]
```diff
@@ -39,6 +40,9 @@ publish lalrpop-util
 publish lalrpop-snap
 publish lalrpop
 
+git tag $VERSION
```
Comment (ruuda) [inline publish.sh]: You should create an [annotated tag](https://git-scm.com/docs/git-tag#_description) with `-a`.

File: publish.sh [hunk 2]
```diff
@@ -39,6 +40,9 @@ publish lalrpop-util
...
+git push origin $VERSION
```
Comment (ruuda) [inline publish.sh]: I think this should be [`git push origin tag $VERSION`](https://git-scm.com/docs/git-push#git-push-ltrefspecgt82308203). Alternatively you can `git push origin --tags`, but that will push *all* tags, which might not be desirable depending on your local repository state.

Comment (ahmedcharles) [reply publish.sh]: I think pushing only one is preferred. Added the `tag` option.

File: publish.sh (hunk 1 above)
Comment (ahmedcharles) [reply publish.sh]: The current tags aren't annotated and they don't provide any benefit I can tell unless they are signed or given a message. The entire point of automating it is so Niko doesn't have to do manual steps and adding a default message seems pointless if it contains no more information than the tag itself.

Comment (ruuda) [reply publish.sh]: For some reason I was under the impression that lightweight tags cannot be pushed, but that is not true, they can. One small benefit of annotated tags is that `git describe` does not ignore them by default. `git describe` can be somewhat useful in automation, but as the version number in `Cargo.toml` needs to be updated anyway, I think you are right, there is little benefit in annotated tags for this use case.

**Pull request merged**